### PR TITLE
Bugfix/cc 18 user login status expire after one week

### PR DESCRIPTION
**Pull request by jennawan**
image
When the user refreshes the page after the expiration period (one week), the "UserInfo" with tokens information in the local storage will be removed and the "Sign out" button will become the anonymous avatar. The same scenario for google login, email login and email login then connect with google.

**Hint for testing**: modify the expiresIn value of back-end, CC-server PR CC-18, @src/auth/auth.service.ts line **264** and **269** to a shorter time.
```
    const atExp = {
      secret: process.env.ACCESS_TOKEN_SECRET,
      expiresIn: 60 * 15, //15 min expiration
    };

    const rtExp = {
      secret: process.env.REFRESH_TOKEN_SECRET,
      expiresIn: 60 * 60 * 24 * 7, //one week expiration
    };
```

### CODE REVIEW

**Review by zhuocun — APPROVED** [review]
Great job!

File: src/auth/auth.service.ts [hunk 1]
```diff
@@ -111,10 +119,10 @@ export class AuthService {
     return respond;
   }
 
-  async userLogout(userId: ObjectId) {
-    const user = await this.userModel.findById(userId);
+  async userLogout(body) {
+    const user = await this.userModel.findById(body.userId);
     user.hashedRefreshToken &&
-      (await this.userModel.findByIdAndUpdate(userId, { hashedRefreshToken: null }));
+      (await this.userModel.findByIdAndUpdate(body.userId, { hashedRefreshToken: null }));
```
Comment (tatoMa) [inline src/auth/auth.service.ts:125]: A nitpick. Destructuring userId? It seems like the only variable you need is userId.

File: src/auth/auth.service.ts [hunk 2]
```diff
@@ -48,23 +48,31 @@ export class AuthService {
         lastName: family_name,
         isActivated: true,
       });
+      // get access token and refresh token
+      const ATandRT = await this.getTokens(newUser._id, newUser.email);
```
Comment (Chanran) [inline src/auth/auth.service.ts:52]: What's the meaning of "ATandRT"？

File: src/auth/auth.service.ts [hunk 3]
```diff
@@ -246,9 +254,9 @@ export class AuthService {
     return tokens;
   }
 
-  async getTokens(adminId: ObjectId, email: string) {
+  async getTokens(userId: ObjectId, email: string) {
```
Comment (Chanran) [inline src/auth/auth.service.ts:257]: UserId or ClientId？

Comment (jennawan) [reply src/auth/auth.service.ts:267]: Changed to UserId for consistency. Modified GetCurrentClient to GetCurrentUser as well.

File: src/auth/auth.service.ts (hunk 2 above)
Comment (jennawan) [reply src/auth/auth.service.ts]: Changed variable name to tokens

File: src/auth/auth.service.ts (hunk 1 above)
Comment (jennawan) [reply src/auth/auth.service.ts]: Thanks for the suggestion. Done!

**Review by tatoMa — APPROVED** [review]
LGTM

**Review by Chanran — APPROVED** [review]
LGTM

**Pull request merged**